### PR TITLE
[Docs] Update prometheus.exporter.* Example(s) sections

### PR DESCRIPTION
**Pull request by spartan0x117**
#### PR Description
- Updates `Example(s)` sections of prometheus exporter flow component documentation. Each example was updated to be a full pipeline, so should be copy-pastable with only a few modifications necessary to test out (replacing URLs, usernames, passwords, etc...).
- Updates the instructions on writing documentation for exporter components accordingly.
- riverfmt each example.

#### Notes to the Reviewer
Open questions that I'd love some feedback on:
- What are the thoughts on the format? 
- For the `Usage` section, should we include a `Replace EXAMPLE_PLACEHOLDER with your intended placeholder.` My initial thinking was no, as each argument and block is covered in detail below.
- Again for the `Usage` section. For any components that don't have any required arguments, applying `grafana-agent fmt` would make them look like: `prometheus.exporter.unix { }`. Is that desired or should they be left with multi-line empty blocks?
- Should the example of the component in question also contain placeholders, rather than "realistic" example values (my thinking was again that the descriptions are above, though perhaps it's confusing)?

#### PR Checklist

- [ ] CHANGELOG updated
- [X] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by erikbaranowski — COMMENTED** [review]
This looks like a great update to our docs.

I'll let @clayton-cornell give any recommendations about the best way to format the new variable docs. I added a couple comments from my read through.

File: docs/sources/flow/reference/components/prometheus.exporter.postgres.md [hunk 1]
```diff
@@ -112,28 +112,37 @@ prometheus.exporter.postgres "example" {
...
+    url = PROMETHEUS_REMOTE_WRITE_URL
+
+    basic_auth {
+      username = USERNAME
+      password = PASSWORD
+    }
   }
 }
 ```
+Replace the following:
```
[the remark this answers is not in the document]
Comment (spartan0x117) [reply docs/sources/flow/reference/components/prometheus.exporter.postgres.md:129]: It's something I'm definitely open to discussion about, but I did intentionally leave out the repeated explanations for the following examples for brevity.

Probably the biggest argument I can think of to repeat the explanations for each example is if we expect that readers jump to specific examples, they might miss the explanations. Do you think we should add the placeholder explanations for each example or only include it for the first occurrence of a placeholder?

File: docs/sources/flow/reference/components/prometheus.exporter.blackbox.md [hunk 2]
```diff
@@ -103,54 +96,80 @@ debug information.
...
+  - `PROMETHEUS_REMOTE_WRITE_URL`: The URL of the Prometheus remote_write-compatible server to send metrics to.
+  - `USERNAME`: The username to use for authentication to the remote_write API.
+  - `PASSWORD`: The password to use for authentication to the remote_write API.
 
 This example is the same above with using an embedded configuration:
 
 ```river
 prometheus.exporter.blackbox "example" {
-    config = "{ modules: { http_2xx: { prober: http, timeout: 5s } } }"
+  config = "{ modules: { http_2xx: { prober: http, timeout: 5s } } }"
```
[the remark this answers is not in the document]
Comment (spartan0x117) [reply docs/sources/flow/reference/components/prometheus.exporter.blackbox.md:145]: Sadly this is a string, so can't be formatted in multiple lines in the config itself :crying_cat_face:

File: docs/sources/flow/reference/components/prometheus.exporter.postgres.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (clayton-cornell) [reply docs/sources/flow/reference/components/prometheus.exporter.postgres.md:129]: There's a saying in the Tech Writing world: "Every page is page one." 

The guideline I go by is... always define things at the first occurrence on a page. If the readers will skip around in the content, page is really long, or there are new replacement variables in the 2nd, 3rd etc examples then redefine them all at each instance. I'd only skip if there are examples one after the other without descriptive content breaking up the information flow.

**Review by clayton-cornell — APPROVED** [review]
Other than possibly adding in the "missing" variable definitions, this all looks good to me. Nothing jumps out as a problem, and... if we need to, we could iterate to tidy up something we've missed.

The multiple variable definitions in a single topic isn't a blocker. We can try it either way and revisit it later it's not enough/too much.

**Review by erikbaranowski — APPROVED** [review]
LGTM 🎉

**Pull request merged**